### PR TITLE
refactor: find new columns to improve write performance

### PR DESCRIPTION
**Pull request by chunshao90**
## Related Issues
Closes #

## Detailed Changes
Refactor find new columns to improve write performance.
* Skip building column schema when columns already exists.

## Test Plan 
TSBS.

## Benchmark
|                | auto_create_table = false | auto_create_table = true| performance loss |
|:--------:|:-------------------------:|:------------------------:|:-----------------:|
| Before    |       171273.03 rows/s        |         152436.35 rows/s    |        11%                |
| After       |       171699.82 rows/s        |        164791.73  rows/s    |       4.1%                |

### CODE REVIEW

File: proxy/src/write.rs [hunk 1]
```diff
@@ -668,32 +658,95 @@ impl<Q: QueryExecutor + 'static> Proxy<Q> {
...
 ) -> Result<Vec<ColumnSchema>> {
-    let new_schema = build_schema_from_write_table_request(schema_config, write_req)
+    let WriteTableRequest {
+        table,
+        field_names,
+        tag_names,
+        entries: write_entries,
+    } = write_table_req;
+
+    let mut columns: BTreeMap<_, ColumnSchema> = BTreeMap::new();
```
Comment (jiacai2050) [inline proxy/src/write.rs:670]: Why require ordering here?

File: proxy/src/write.rs [hunk 2]
```diff
@@ -668,32 +658,95 @@ impl<Q: QueryExecutor + 'static> Proxy<Q> {
...
+                let field_index = field.name_index as usize;
+                ensure!(
+                    field_index < field_names.len(),
+                    InternalNoCause {
+                        msg: format!(
+                        "Field {field:?} is not found in field_names:{field_names:?}, table:{table}",
+                    ),
+                    }
+                );
+                if (field.name_index as usize) < field_names.len() {
```
Comment (jiacai2050) [inline proxy/src/write.rs:701]: Doesn't this check duplicated with ensure?

File: proxy/src/write.rs [hunk 3]
```diff
@@ -668,32 +658,95 @@ impl<Q: QueryExecutor + 'static> Proxy<Q> {
 
 fn find_new_columns(
```
Comment (jiacai2050) [inline proxy/src/write.rs:659]: Better to write a unit test for this function.

File: proxy/src/write.rs [hunk 4]
```diff
@@ -668,32 +658,95 @@ impl<Q: QueryExecutor + 'static> Proxy<Q> {
...
+
+fn build_column<'a>(
+    columns: &mut BTreeMap<&'a str, ColumnSchema>,
+    schema: &Schema,
+    name: &'a str,
+    value: &Option<Value>,
+    is_tag: bool,
+) -> Result<()> {
+    // Skip adding columns, the following cases:
+    // 1. Field already exists.
```
Comment (jiacai2050) [inline proxy/src/write.rs:720]: Column already exists.

**Comment by ShiKaiWi** [comment]
@chunshao90 Could you add the unit for the benchmark results (I guess it should be rows/s)?

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**